schedule/jeos/sle/jeos-main.yaml

```diff
@@ -38,8 +38,8 @@ schedule:
     - console/journal_check
     - microos/libzypp_config
     - console/suseconnect_scc
-    - '{{maintenance}}'
     - '{{efi}}'
+    - '{{maintenance}}'
```

Comment:
That is actually the catch, it depends on whether `update-bootloader --reinit` has been triggered during the update process. So we would either need to determine what has been updated or whether the SUT already booted from sle/opensuse efiboot record.

Reply:
> That is actually the catch, it depends on whether update-bootloader --reinit has been triggered during the update process.

Ah, I see. The failing test linked in the ticket calls `shim-install` and reboots before the failure happens, so it didn't look that way to me. What exactly causes that particular failure?

> So we would either need to determine what has been updated

If it's about `shim-install`, then checking `efibootmgr` for a `-secureboot` entry should be sufficient. Does it actually make a difference whether the system was booted through a boot entry or the removable location? Though `patch_and_reboot` does a reboot, so if the entry exists it was also booted through it.

Reply:
> > That is actually the catch, it depends on whether update-bootloader --reinit has been triggered during the update process.
> 
> Ah, I see. The failing test linked in the ticket calls `shim-install` and reboots before the failure happens, so it didn't look that way to me. What exactly causes that particular failure?
> 

Sorry, I have missed this one in the ticket. :facepalm:  It is a race condition in openqa, where the test should enroll a mock certificate using MOK https://openqa.suse.de/tests/6380111#step/verify_efi_mok/246. 

However, sometimes it misses a needle and does not interrupt the MOK timeout countdown. The MOK is always visible in serial line though https://openqa.suse.de/tests/6382793/logfile?filename=serial0.txt

```
Press any key to perform MOK managementBooting in 10 seconds  Booting in 9 seconds  Booting in 8 seconds  Booting in 7 seconds  Booting in 6 seconds  Booting in 5 seconds  Booting in 4 seconds  Booting in 3 seconds  Booting in 2 seconds  Booting in 1 second   UEFI SHIM
$Version: 15 $
$BuildMachine: Linux x86_64 x86_64 x86_64 GNU/Linux $
$Commit: master $
Verification succeeded
Welcome to GRUB!
```

> > So we would either need to determine what has been updated
> 
> If it's about `shim-install`, then checking `efibootmgr` for a `-secureboot` entry should be sufficient. Does it actually make a difference whether the system was booted through a boot entry or the removable location? Though `patch_and_reboot` does a reboot, so if the entry exists it was also booted through it.

I cannot recall to run the test against SUT with removable drive. It runs in some maintenance installation scenario, but after the system is deployed, thus `efibootmgr` shows the boot entry.

Reply:
> I cannot recall to run the test against SUT with removable drive.

On the first boot on any image (HDD, ISO, USB) there aren't any boot entries configured in the EFI, so it uses the `/EFI/BOOT/bootx64.efi` executable, which is called "removable" because it's most commonly used on removable media.

Reply:
> Sorry, I have missed this one in the ticket. facepalm It is a race condition in openqa, where the test should enroll a mock certificate using MOK https://openqa.suse.de/tests/6380111#step/verify_efi_mok/246.

How does the failure actually look like when `patch_and_reboot` happens before `verify_efi_mok`?

Reply:
> > I cannot recall to run the test against SUT with removable drive.
> 
> On the first boot on any image (HDD, ISO, USB) there aren't any boot entries configured in the EFI, so it uses the `/EFI/BOOT/bootx64.efi` executable, which is called "removable" because it's most commonly used on removable media.

Thanks! So then the test expects "removable" whenever it comes to JeOS. It misses the part that installing updates could possibly set the boot entries.

Reply:
> > Sorry, I have missed this one in the ticket. facepalm It is a race condition in openqa, where the test should enroll a mock certificate using MOK https://openqa.suse.de/tests/6380111#step/verify_efi_mok/246.
> 
> How does the failure actually look like when `patch_and_reboot` happens before `verify_efi_mok`?

The closest what I have found. It matches the first MOK manager needle, but still fails to cancel the timeout.
https://openqa.suse.de/tests/6380133#step/verify_efi_mok/246

Reply:
I don't get how this could be influenced by the order of test modules then

Reply:
> I don't get how this could be influenced by the order of test modules then

I am a bit omitting the MOK manager failure in this PR. The focus is to enable this module to work with updates regardless whether boot records have been touched, and to make it work on `aarch64` for SP3.
Sorry for the mess!

Reply:
What I'm missing is a connection between the changes in this PR and the test failures. Currently my impression is that there's a race which openQA somehow wins if the order of modules is changed?

Reply:
> What I'm missing is a connection between the changes in this PR and the test failures. Currently my impression is that there's a race which openQA somehow wins if the order of modules is changed?

The ticket is not that helpful, sorry. There are 3 different failures, let me try to collect them when today's update tests will finish.

Reply:
I have updated the ticket with all failures that appeared in the test case lately. If I remember correctly, [bootloader-shim-verification needle](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/opensusebasetest.pm#L683) was used in o3, however it was not matched for 9 months and the test run is gone, so I am not sure whether there was a change in the product or what has happened. Do you know if this specific needles is still needed, please ? It would make it easier to update the handling MOK Manager branch.

Reply:
This one is still used https://openqa.opensuse.org/tests/1839636#step/bootloader_uefi/1

Reply:
So removing the check for `/proc/keys` is to workaround the needle matching issues on booting aarch64 systems?

Reply:
> So removing the check for `/proc/keys` is to workaround the needle matching issues on booting aarch64 systems?

The workaround is to scan serial output whether Mok Manager has been invoked. I am not confident if `/proc/keys` check is necessary in this test case.

Reply:
Eventually, it seems to be a bug https://bugzilla.suse.com/show_bug.cgi?id=1182211#c31